Tests/Marketplace/search_and_install_packs.py

```diff
@@ -372,7 +374,16 @@ def search_pack_and_its_dependencies(client: demisto_client,
 
         current_packs_to_install = [pack_data]
         if dependencies:
-            current_packs_to_install.extend(dependencies)
+            # Check that the dependencies don't include a deprecated pack:
+            for dependency in dependencies:
+                pack_path = os.path.join(PACKS_DIR, dependency.get('id'))
+                if is_pack_deprecated(pack_path):
```

Comment:
why we don't pass pack_data to `is_pack_deprecated`, isn't the data contains the metadata ?
passing the path, means opening and loading files each time this function is called just to get single field

Reply:
@anara123 The `pack_data` returned from the `search_pack()` function includes only ID and version fields. It doesn't include the metadata of the pack.
![image](https://user-images.githubusercontent.com/61732335/105809949-c6428e00-5fb2-11eb-80d6-d36633c88f6c.png)

Reply:
@anara123, @darkushin  we could add somewhere a file with all deprecated \ hidden packs and cache it so that all threads can just read it.

Maybe adding this info to the id_set where the id is the pack display name, any thoughs?

Reply:
To check support we do the same thing as well. We can add a function that will create some kind of a cache but what will we save? time? I am not sure that building this cache upfront for all packs will be faster than performing the file reading operation a couple of times for each pack.

Reply:
I think we can make this an enhancement If we will see that it extends the build time.

Reply:
Agreed, but let's check it before we merge to know better what to expect.

I would like to compare a build that installs say 20 packs that have non-trivial dependencies and see how much that adds to the build and then lets decide how much we want to prioritize it.

@bakatzir have just reduced the sdk build time by 8 minutes only by removing the repeating reading and parsing of the id_set file.

Reply:
BTW I see we are not using this flow in the nightly, is this on porpuse? @darkushin

Reply:
@hod-alpert 
as long as we are installing all of the packs in the bucket in the nightly, why should we filter?

Reply:
@hod-alpert We do filter in nightly:
https://github.com/demisto/content/pull/10952/files#diff-25b80cc78a0efaee708db1ce39fb154ac3358eae6361e3e336922d8a77599863R1250

Reply:
@ShahafBenYakir we are filtering, but we are not checking for every pack that all its dependencies are not deprecated. So, for example, if pack A is deprecated and pack B depends on pack A, we will filter pack A but pack B will still be installed in nightly. Is it ok?

Reply:
I think Pack B will not be installed. The operation will fail as it misses its dependency. no @hod-alpert?

Reply:
Yes, the entire build will fail and it will be written in the logs that pack B misses its dependency pack A. What I meant is that in nightly we will collect pack B and try to install it (but fail), whereas in a normal build we won't try to install pack B in the first place. I think it is ok this way.

Reply:
In that case I'm pretty sure the nightly will try to install it, fail and then try again without the problematic pack, it;s a feature @Shellyber implemented.